presto-native-execution/presto_cpp/main/operators/tests/UnsaferowShuffleTest.cpp

```diff
@@ -264,57 +352,96 @@ TEST_F(UnsafeRowShuffleTest, operators) {
   });
+  runShuffleTest(&shuffle, numPartitions, numMapDrivers, {data});
+}
 
-  auto dataType = asRowType(data->type());
+TEST_F(UnsafeRowShuffleTest, persistentShuffle) {
+  size_t numPartitions = 5;
+  size_t numMapDrivers = 2;
 
-  auto leafPlan = exec::test::PlanBuilder()
-                      .values({data}, true)
-                      .addNode(addPartitionAndSerializeNode(numPartitions))
-                      .localPartition({})
-                      .addNode(addShuffleWriteNode(&shuffle))
-                      .planNode();
+  // Create a local file system storage based shuffle.
+  velox::filesystems::registerLocalFileSystem();
+  auto rootDirectory = velox::exec::test::TempDirectoryPath::create();
+  auto rootPath = rootDirectory->path;
 
-  auto leafTaskId = makeTaskId("leaf", 0);
-  auto leafTask = makeTask(leafTaskId, leafPlan, 0);
-  exec::Task::start(leafTask, 2);
-  ASSERT_TRUE(exec::test::waitForTaskCompletion(leafTask.get()));
-
-  // Create and run multiple downstream tasks, one per partition, to read data
-  // from shuffle.
-  for (auto i = 0; i < numPartitions; ++i) {
-    auto plan = exec::test::PlanBuilder()
-                    .exchange(dataType)
-                    .project(dataType->names())
-                    .planNode();
-
-    exec::test::CursorParameters params;
-    params.planNode = plan;
-    params.destination = i;
-
-    bool noMoreSplits = false;
-    auto [taskCursor, serializedResults] = readCursor(params, [&](auto* task) {
-      if (noMoreSplits) {
-        return;
-      }
-      addRemoteSplits(task, {leafTaskId});
-      noMoreSplits = true;
-    });
+  // Initialize persistent shuffle.
+  TestingPersistentShuffle shuffle(1 << 20 /* 1MB */);
+  shuffle.initialize(pool(), numPartitions, rootPath);
+
+  auto data = vectorMaker_.rowVector({
+      makeFlatVector<int32_t>({1, 2, 3, 4, 5, 6}),
+      makeFlatVector<int64_t>({10, 20, 30, 40, 50, 60}),
+  });
+  runShuffleTest(&shuffle, numPartitions, numMapDrivers, {data});
+}
 
-    ASSERT_FALSE(shuffle.hasNext(i)) << i;
+TEST_F(UnsafeRowShuffleTest, persistentShuffleFuzz) {
+  // For unit testing, these numbers are set to relatively small values.
+  // For stress testing, the following parameters and the fuzzer vector,
+  // string and container sizes can be bumped up.
+  size_t numPartitions = 5;
+  size_t numMapDrivers = 2;
+  size_t numInputVectors = 5;
+  size_t numIterations = 5;
+
+  // Set up the fuzzer parameters.
+  VectorFuzzer::Options opts;
+  opts.vectorSize = 1000;
+  opts.nullRatio = 0.1;
+  opts.containerHasNulls = false;
+  opts.dictionaryHasNulls = false;
+  opts.stringVariableLength = true;
+  // UnsafeRows use microseconds to store timestamp.
+  opts.useMicrosecondPrecisionTimestamp = true;
+  opts.stringLength = 100;
+  opts.containerLength = 10;
+
+  // For the time being, we are not including any MAP or more than three level
+  // nested data structures given the limitations of the fuzzer and
```

Comment:
> limitations of the fuzzer and assertEqualResults

It would be nice to add links to corresponding GitHub issues.

Also, I don't see BIGINT type in the list. Any particular reason it is missing?

Reply:
Done